File: CS50x/pset6/readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    countL = count_letters(text)
    countW = count_words(text) + 1
    countS = count_sentences(text)

    L = (countL * 100) / countW
    S = (countS * 100) / countW

    grade = round(0.0588 * L - 0.296 * S - 15.8)

    if grade < 1:
        print("Before Grade 1")
    elif grade > 16:
        print("Grade 16+")
    else:
        print(f"Grade: {grade}")


def count_letters(text):
    count = 0
    for l in text:
        if l.lower() in ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z']:
            count += 1
    return count


def count_words(text):
    count = 0
    for l in text:
        if l in [' ', '\n']:
            count += 1
    return count


def count_sentences(text):
    count = 0
    for l in text:
        if l in ['.', '?', '!']:
            count += 1
    return count


if __name__ == '__main__':
    main()

File: CS50x/pset2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //get input from user
    string text = get_string("Text: ");

    //analyze the text
    int NumLet = count_letters(text);
    int NumWrd = count_words(text);
    int NumSen = count_sentences(text);

    //get the grade
    float L = (NumLet * 100) / (float) NumWrd;
    float S = (NumSen * 100) / (float) NumWrd;
    int grade = (int) round(0.0588 * L - 0.296 * S - 15.8);

    //print the grade
    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }

}


int count_sentences(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        char c = text[i];
        if (c == '.' || c == '!' || c == '?')
        {
            count++;
        }
    }
    return count;
}

int count_words(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isspace(text[i]))
        {
            count++;
        }
    }
    return count + 1;
}

int count_letters(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (isalpha(text[i]))
        {
            count++;
        }
    }
    return count;
}